201-front-door-standard-premium-app-service-public/main.bicep

```diff
@@ -0,0 +1,175 @@
+param location string = resourceGroup().location
+param appName string
+param appServicePlanSkuName string
+param appServicePlanCapacity int
+param frontDoorEndpointName string
+param frontDoorSkuName string {
+  allowed: [
+    'Standard_AzureFrontDoor'
+    'Premium_AzureFrontDoor'
+  ]
+}
+
+var appServicePlanName = 'AppServicePlan'
+
+var frontDoorProfileName = 'MyFrontDoor'
+var frontDoorOriginGroupName = 'MyOriginGroup'
+var frontDoorOriginName = 'MyAppServiceOrigin'
+var frontDoorOriginPath = ''
+var frontDoorRouteName = 'MyRoute'
+
+resource frontDoorProfile 'Microsoft.Cdn/profiles@2020-09-01' = {
+  name: frontDoorProfileName
+  location: 'global'
+  sku: {
+    name: frontDoorSkuName
+  }
+  properties: {}
+}
+
+resource appServicePlan 'Microsoft.Web/serverFarms@2020-06-01' = {
+  name: appServicePlanName
+  location: location
+  sku: {
+    name: appServicePlanSkuName
+    capacity: appServicePlanCapacity
+  }
+  kind: 'app'
+}
+
+resource app 'Microsoft.Web/sites@2020-06-01' = {
+  name: appName
+  location: location
+  kind: 'app'
+  properties: {
+    serverFarmId: appServicePlan.id
+    httpsOnly: true
+    siteConfig: {
+      ipSecurityRestrictions: [
+        {
+          tag: 'ServiceTag'
+          ipAddress: 'AzureFrontDoor.Backend'
+          action: 'Allow'
+          priority: 100
+          headers: {
+            'x-azure-fdid': [
+              frontDoorProfile.properties.frontDoorId
+            ]
+          }
+          name: 'Allow traffic from Front Door'
+        }
+      ]
+    }
+  }
+}
+
+resource frontDoorEndpoint 'Microsoft.Cdn/profiles/afdEndpoints@2020-09-01' = {
+  name: '${frontDoorProfile.name}/${frontDoorEndpointName}'

```

Comment:
is the parent: syntax for child resource done yet?  if so we should use it...

Reply:
This isn't available yet (the Bicep PR is still open). I planned to do an update to all of these samples once Bicep has a few more features, to bring them in line with the updated best practices. However I would like to get these samples published soon (rather than waiting for Bicep to be completely ready) as they are going to be in the Front Door docs.